src/main/java/com/example/onlineservicesemulator/Controller.java

```diff
@@ -2,44 +2,87 @@
 
 
 import com.example.onlineservicesemulator.classes.JSONReader;
+import javafx.event.EventHandler;
 import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
 import javafx.fxml.Initializable;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
 import javafx.scene.control.CheckBox;
 import javafx.scene.control.Label;
+import javafx.scene.control.ListView;
+import javafx.scene.input.MouseEvent;
 import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
 
+import java.io.File;
+import java.io.IOException;
 import java.net.URL;
-import java.util.List;
-import java.util.ResourceBundle;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
 
 public class Controller implements Initializable {
 
+    @FXML
+    private ListView<String> filesList;
     @FXML
     private VBox checkboxList;
+    @FXML
+    private Button addFileButton;
+    @FXML
+    private Button removeFileButton;
+    private List<String> servicesNames;
+    private Map<String, List<String>> servicesAndUploadedFilesMap;
 
     @Override
     public void initialize(URL url, ResourceBundle resourceBundle) {
+        servicesNames = JSONReader.getServices();
         createAndSetCheckboxList();
+        setFilesMap();
+        addFileButton.setVisible(false);
+        setRemoveFileButtonListener();
     }
 
     public void createAndSetCheckboxList() {
-        List<String> servicesNames = JSONReader.getServices();
-        double lenghtSectionA = calculateLongestServiceString(servicesNames);
-        checkboxList.setPrefWidth(lenghtSectionA);
+        double lengthSectionA = calculateLongestServiceString(servicesNames) * 8;
```

Comment:
can be int instead of double, lets pray to god that there is no service requiring the capacity for double, that would require a very long name, or to be even more space efficient, this could even be a short